### ISSUE TITLE
Release needed to solve problems in Fiji

### ISSUE THREAD

**Issue by tferr** [scijava/scijava-plot#5]
@ctrueden, @maarzt,

SNT uses plot-service extensively (I am always surprised we are the only ones!?), but we keep getting reports that SNT injects a Table root menu. This menu hosts broken plugins declared in imagej-plot-service
![image](https://user-images.githubusercontent.com/2439948/162005153-ab7cddc2-bfbb-47a4-aa24-ddbc73cc2a81.png)

Where do this plugins come from?
The imagej-plot-service.jar contains them under `/tableplugins/`, but I cannot find their source code!?

- Would it be possible to remove them, or patch them?
- These undocumented plugins don't seem to (the logic to retrieve columns in their dialog prompts seems to be broken)

Also, what is the state of the plot-service package? It somehow feels like a project that is not official. Should we not use it? Should it move to a more officialized organization?

EDIT: Initially I thought updating to v0.1.2-SNAPSHOT would fix this issue, but I just realize it does not. Updated this, accordingly.

**Comment by ctrueden** [comment]
> It somehow feels like a project that is not official.

Yeah, the plan is to merge it into scijava-common. I have just failed to review the code, even after years, because I'm always too busy. 😞

@tferr You could help me out here, since you have been using it: are you completely satisfied with the API and functionality? Do you think we could merge it into scijava-common completely as is? Or are there changes you would suggest to make before we do so?

**Comment by ctrueden** [comment]
As for the mysterious `/tableplugins/`, and `imagej-plot-service.jar`, you can find out where a JAR came from by using the System Information command inside Fiji. Copy and paste the result into a text editor of your choice, then search it for imagej-plot-service and see what you find.

Or if that doesn't work, you can:

```
mkdir t
$ jar xf /path/to/imagej-plot-service-x.y.z.jar
$ cat META-INF/MANIFEST.MF | grep Implementation
Implementation-Title: SciJava Common
Implementation-Version: 2.80.1
Implementation-Date: 2019-11-01T19:35:11+0000
Implementation-Vendor: SciJava
Implementation-Build: 76dc26b325cf84d40d7631014085175f560c0954
$ grep scm META-INF/maven/*/*/pom.xml
	<scm>
		<connection>scm:git:https://github.com/someOrg/imagej-plot-service</connection>
		<developerConnection>scm:git:git@github.com:someOrg/imagej-plot-service</developerConnection>
	</scm>
$ cd ..
$ rm -rf t
```

**Comment by tferr** [comment]
> @tferr You could help me out here, since you have been using it: are you completely satisfied with the API and functionality? Do you think we could merge it into scijava-common completely as is? Or are there changes you would suggest to make before we do so?

I think the API for XY plots (scatter and line) is OK as is. For those plots, we use plot service exclusively (I think this is  always true, but will have to double-check). I have less experience with other types of plots (histograms, box plot, etc.). For those we decided to use jfreechart directly, because 1) plot service adopts the default theme of jfreechart that I personally feel is not very "Scientific" (most of the time, this boils down to aesthetics), and 2) the plot service API lacks bells and whistles for tweaking such category charts into 'scientific charts'. Most folks may not care about this.

If I understood correctly, the plot service has been designed to be generic and simple. So those bells and whistles are not be provided here anyway. So in short, I think it is probably fine as is. The customizations we use in SNT are all [here](https://github.com/morphonets/SNT/blob/d55b8523a94de0cba3050d3c9ed703ad710b7e65/src/main/java/sc/fiji/snt/analysis/SNTChart.java), so in the future if others feel that the default theme should change, we could bring the customizations there into Plotservice (without changing the API).

If you don't have time, I don't mind maintaining the status quo. Currently, the only issues is that Table>  menu seems to be annoying folks everywhere that blame us, and we don't have anything to do with it :)

I did some digging. Those  Table plugins used to exist until [d3c749](https://github.com/maarzt/scijava-plot/tree/d3c749ae662c1a84731cc721412184b270291200/src/main/java/tableplugins). After that they were removed from master. It seems that there was no build since then, Since anyway, they are not working, if a new release is made with the current code in master, the menu will disappear, and the issue is fixed?

**Comment by tferr** [comment]
I think the problem is also, that there was never a release since this was renamed.
So https://maven.scijava.org/, lists only imagej-plot-service (that registers the funky Tables menu), 
This package (that no longer has the Table> menu) is now `scijava-plot` but is not listed on maven.

So, I'm now certain that releasing a new version (available in Maven) will fix this. I can try to open a PR, in case it helps

**Comment by ctrueden** [comment]
@maarzt How about transferring this repository into the scijava org? Then we can cut a new release!

**Comment by maarzt** [comment]
@ctrueden I transferred the repository to the scijava organization. Feel free to cut the release.

I don't have an opinion of whether to merge it with scijava-common or not. Both is fin with me.

**Comment by maarzt** [comment]
Other thought's about this repo:
1. @tferr I would love to have the charts look scientific :D Could you setup a PR, that changes the style of the plot service charts as you like them?
2. The PlotService API only has the one JFreeChart backend. Not sure if the interface works well also for another backed. It would be interesting to try chart-fx as backend.

**Comment by ctrueden** [comment]
* I released [scijava-plot 0.2.0](https://maven.scijava.org/#nexus-search;gav~org.scijava~scijava-plot~0.2.0~~).
* I merged @maarzt's `plot-service` branch on his scijava-ui-swing fork to the scijava-ui-swing mainline, and released [scijava-ui-swing 0.14.0](https://maven.scijava.org/#nexus-search;gav~org.scijava~scijava-ui-swing~0.14.0~~).
* I updated pom-scijava to include scijava-plot and the new scijava-ui-swing: scijava/pom-scijava@c9df9fd49768cb7bd2462489d01ae61bffa321a7; it will be part of the pom-scijava 32.0.0 release.

@tferr Once pom-scijava 32 is released and Fiji is updated to match, you won't have to ship scijava-plot or scijava-ui-swing on your update site anymore.

**Comment by tferr** [comment]
> * @tferr I would love to have the charts look scientific :D Could you setup a PR, that changes the style of the plot service charts as you like them?

Currently, I don't have immediate time to look into this. I opened an [issue](https://github.com/morphonets/SNT/issues/104) on our project so that it is not forgotten, it includes notes on where our customizations reside, in case someone can look into it in the interim. To be clear, with 'scientific', I meant more 'publication-like', 'formal' layouts, see e.g., this [snapshot](https://github.com/morphonets/SNT/issues/104#issue-1196708616).

> * The PlotService API only has the one JFreeChart backend. Not sure if the interface works well also for another backed. It would be interesting to try chart-fx as backend.

It would, I did not pursue it much at the time, because my understanding was that using java-fx in Fiji was somewhat problematic (Maybe that is no longer the case!?). BTW, I also think there is a new major version of jfreechart coming out soon, so testing that will also be interesting.
